### PR TITLE
Update PyPy in CI to 3.10

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview

* Add PyPy3.10 to CI image
* Fix a broken settings test that seems to throw errors on PyPy3.10

### CODE REVIEW

File: .github/containers/Dockerfile [hunk 1]
```diff
@@ -96,7 +96,7 @@ RUN echo 'eval "$(pyenv init -)"' >>${HOME}/.bashrc && \
     pyenv update
 
 # Install Python
-ARG PYTHON_VERSIONS="3.11 3.10 3.9 3.8 3.7 3.12 2.7 pypy2.7-7.3.12 pypy3.8-7.3.11"
+ARG PYTHON_VERSIONS="3.11 3.10 3.9 3.8 3.7 3.12 2.7 pypy2.7-7.3.12 pypy3.10-7.3.15 pypy3.8-7.3.11"
```
Comment (hmstepanek) [inline .github/containers/Dockerfile:99]: I think we can just drop pypy3.8 here now can't we?

File: tests/mlmodel_sklearn/test_cluster_models.py [hunk 2]
```diff
@@ -19,10 +19,10 @@
 )
 
 from newrelic.api.background_task import background_task
```
Comment (hmstepanek) [inline tests/mlmodel_sklearn/test_cluster_models.py:21]: These changes are in another PR so let's just remove the sklearn changes from this PR.

File: tests/agent_unittests/test_utilization_settings.py [hunk 3]
```diff
@@ -118,7 +118,7 @@ def reset(wrapped, instance, args, kwargs):
     return reset
 
 
-@reset_agent_config(INI_FILE_WITHOUT_UTIL_CONF, ENV_WITHOUT_UTIL_CONF)
+@reset_agent_config(INI_FILE_WITHOUT_UTIL_CONF, {"NEW_RELIC_HOST": "collector.newrelic.com"})
```
Comment (hmstepanek) [inline tests/agent_unittests/test_utilization_settings.py:121]: It's concerning to me that the HOST is needed here. It shouldn't need the host to default the OTLP_HOST. Can you explain more about this change/why this test was failing? I just tried running it locally and it didn't fail for me.

**Review by hmstepanek — COMMENTED** [review]
A couple minor points of feedback. Overall the switch from pypy3.8 to pypy3.10 looks good.

File: .github/containers/Dockerfile (hunk 1 above)
Comment (TimPansino) [reply .github/containers/Dockerfile:99]: I think there might be at least 1 framework that doesn't support pypy3.8, I have 3 left that haven't passed yet with 3.10. Will take another look tomorrow to see if I can get them all to pass and I'll drop 3.8 if that works.

File: tests/agent_unittests/test_utilization_settings.py (hunk 3 above)
Comment (TimPansino) [reply tests/agent_unittests/test_utilization_settings.py:121]: This test is really annoying and fails if you've got NEW_RELIC_HOST set to point to staging. It causes my local instance to fail all the time.

If you've got a better idea to make this test not fail I'd be happy to change it.

File: .github/containers/Dockerfile (hunk 1 above)
Comment (TimPansino) [reply .github/containers/Dockerfile:99]: I think I got them all running now, I dropped 3.8 completely

File: tests/agent_unittests/test_utilization_settings.py (hunk 3 above)
Comment (hmstepanek) [reply tests/agent_unittests/test_utilization_settings.py:121]: I was thinking we had other tests for when this was specifically set that would make this change to it a duplicate test but I don't actually see any so I guess this change is ok.

**Review by hmstepanek — APPROVED** [review]
Looks good!

**Comment by TimPansino** [comment]
Last test run passing, switching the CI image tag back to latest

https://github.com/newrelic/newrelic-python-agent/actions/runs/7982302622?pr=1065

**Pull request merged**